pyneuroml/utils/CellBuilder.py

```diff
@@ -0,0 +1,296 @@
+    )
+    segment = Segment(id=segid, proximal=p, distal=d, parent=sp)
+
+    if name:
+        segment.name = name
+
+    if group:
+        seg_group = None
+        seg_group = cell.get_segment_group(group)
+        seg_group_default = None
+        neuro_lex_id = None
+
+        if "axon_" in group:
+            neuro_lex_id = neuro_lex_ids[
+                "axon"
+            ]  # See http://amigo.geneontology.org/amigo/term/GO:0030424
+            if use_convention:
+                seg_group_default = cell.get_segment_group("axon_group")
+        if "soma_" in group:
+            neuro_lex_id = neuro_lex_ids["soma"]
+            if use_convention:
+                seg_group_default = cell.get_segment_group("soma_group")
+        if "dend_" in group:
+            neuro_lex_id = neuro_lex_ids["dend"]
+            if use_convention:
+                seg_group_default = cell.get_segment_group("dendrite_group")
+
+        if seg_group is None:
+            seg_group = SegmentGroup(id=group, neuro_lex_id=neuro_lex_id)
+            cell.morphology.segment_groups.append(seg_group)
+
+        seg_group.members.append(Member(segments=segment.id))
+        # Ideally, these higher level segment groups should just include other
+        # segment groups using Include, which would result in smaller NML
+        # files. However, because these default segment groups are defined
+        # first, they are printed in the NML file before the new segments and their
+        # groups. The jnml validator does not like this.
+        # TODO: clarify if the order of definition is important, or if the jnml
+        # validator needs to be updated to manage this use case.
+        if use_convention and seg_group_default:
+            seg_group_default.members.append(Member(segments=segment.id))
+
+    if use_convention:
+        seg_group_all = cell.get_segment_group("all")
+        seg_group_all.members.append(Member(segments=segment.id))
+
+    cell.morphology.segments.append(segment)
+    return segment
+
+
+def set_init_memb_potential(cell: Cell, v: str, group: str = "all") -> None:
+    """Set the initial membrane potential of the cell.
+
+    :param cell: cell to modify
+    :type cell: Cell
+    :param v: value to set for membrane potential with units
+    :type v: str
+    :param group: id of segment group to modify
+    :type group: str
+    """
+    cell.biophysical_properties.membrane_properties.init_memb_potentials = [
+        InitMembPotential(value=v, segment_groups=group)
+    ]
+
+
+def set_resistivity(cell: Cell, resistivity: str, group: str = "all") -> None:
+    """Set the resistivity of the cell
+
+    :param cell: cell to modfify
+    :param resistivity: value resistivity to set with units
+    :type resistivity: str
+    :param group: segment group to modify
+    :type group: str
+    """
+    cell.biophysical_properties.intracellular_properties.resistivities = [
+        Resistivity(value=resistivity, segment_groups=group)
```

Comment:
Setting multiple resistivity values for different segment groups is not possible. Only the latest updated resistivity gets reflected in the NeuroML code. It can be rectified by using what I tried locally:

cell.biophysical_properties.intracellular_properties.resistivities.append(Resistivity(value=resistivity, segment_groups=group)

Reply:
Thanks, that's a bug. We should be appending here instead of overwriting the list.